pkg/services/ngalert/api/authorization.go

```diff
@@ -58,6 +58,10 @@ func (api *API) authorize(method, path string) web.Handler {
 			ac.EvalPermission(ac.ActionAlertingRuleCreate, scope),
 			ac.EvalPermission(ac.ActionAlertingRuleDelete, scope),
 		)
+	// Grafana rule state history paths
+	case http.MethodGet + "/api/ruler/grafana/states":
```

Comment:
Open to feedback on this URI, but there were a few careful considerations here:

- It felt cleaner to hang this off the ruler endpoint, as it's logically asking about what happened inside the ruler, in the past.
- `/api/ruler/grafana/states/history` Implies the existence of `/api/ruler/grafana/states` for non-historical states, which we aren't planning on adding.
- An rpc-style `POST .../query-states` or `POST ../query-history` makes creating the label queries easier, but it's not quite as idiomatic.

Reply:
I avoided this approach because the resource hierarchy of `rule/history` implies that's the history for an individual rule resource. But, the endpoint doesn't work like this - you can also issue queries across all the rules.

Reply:
That's a good point - I'm fine with `rules/history`  - I don't think `evaluation` adds anything. IMO `versions` as part of an API Is a common idiom and I would expect something related to versions of a rule to have this keyword in its path as opposed to history but can see the argument.

Reply:
Done! I admit it feels a bit weird not adding this to `/rule/` - my only issue there, is that everything under `/rule/` today is actually just the testing API, and this isn't logically associated with that at all. Maybe those could be moved to `/api/v1/testing/...` at some point?

We could even just do `/api/v1/state-history` which is super safe, and not attached to anything. It's just one endpoint though. :shrug: